### PR TITLE
[server] Pin oci-tool to 0.1.1

### LINKED ISSUE #7174: [vscode-insiders] fix nightly tag resolution
### Bug description

We need to pin oci-tool in server dockerimage and check that it works for VS Code Insiders. After this change is deployed by meta, we will revert pinning of insider image.

### Steps to reproduce

See above.

### Workspace affected

_No response_

### Expected behavior

_No response_

### Example repository

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by geropl**
## Description

/cc @csweichel 

## Related Issue(s)

Fixes #7174

## How to test
 - opne workspace on this branch
 - `kubectl logs server-... -f`
 - note that there are now `oci-tool` related errors anymore:
 ```
{"component":"server","severity":"INFO","time":"2021-12-13T11:34:31.122Z","environment":"devstaging","message":"ide config: successfully resolved image digest","payload":{"ide":"code-latest","image":"eu.gcr.io/gitpod-core-dev/build/ide/code:commit-4cf15538da2ecff8bae9245f18aa3ac6f3b90c3f","resolvedImage":"eu.gcr.io/gitpod-core-dev/build/ide/code:commit-4cf15538da2ecff8bae9245f18aa3ac6f3b90c3f@sha256:d649fa6f7726ea1841928d738b8b5bece0ea516caba11ee31a811f8dacf2500b","trigger":"file changed"}}
```

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

Associated issue: *#7174*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/leeway.Dockerfile
Comment (csweichel) [inline components/server/leeway.Dockerfile:16]: I just pushed a semver compatible tag: v0.1.1 ... now this works:
```
go install github.com/csweichel/oci-tool@v0.1.1
```

Comment (geropl) [reply components/server/leeway.Dockerfile]: Done.

File: components/server/testd/Dockerfile
Comment (csweichel) [inline components/server/testd/Dockerfile:4]:
```suggestion
RUN go install github.com/csweichel/oci-tool@v0.1.1
```

Comment (geropl) [reply components/server/testd/Dockerfile]: Whoops, test file, removed it. Sorry for the noise. :pray:

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: afee8d06898ad741357b925c8e253d037624b10b</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7198#issuecomment-992420183" title="LGTM">csweichel</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

Associated issue: *#7174*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by akosyakov** [comment]
@jeanp413 Could you review it please? that it fixes resolution of nightly tag

**Review by jeanp413 — APPROVED** [review]
LGTM, nightly image now resolves correctly.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7198#issuecomment-992420183" title="LGTM">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7198#pullrequestreview-830329817" title="Approved">jeanp413</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

Associated issue: *#7174*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7198#issuecomment-992420183" title="LGTM">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7198#issuecomment-992559262" title="Approved">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7198#pullrequestreview-830329817" title="Approved">jeanp413</a>*

Associated issue: *#7174*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
